src/Agent.Worker/ExecutionContext.cs

```diff
@@ -19,13 +19,14 @@ public interface IExecutionContext : IAgentService
         TaskResult? Result { get; set; }
         TaskResult? CommandResult { get; set; }
         CancellationToken CancellationToken { get; }
+        Int32 TimeoutInMinutes { get; }
```

Comment:
@vithati, I discussed offline with @TingluoHuang and @patrickcarnahan. Instead of an approach that modifies process invoker, can you do the following:
1) If you need to add a timeout, add it to IStep and please use `System.TimeSpan`.
2) When you create the cancellation token for the step's execution context, create a separate linked cancellation token that is based on the timeout.
3) No modifications to process invoker should be required. An operation canceled exception should bubble out to StepsRunner. Ultimately it seems like we want to distinguish to the end user whether the step failed due to task timeout - versus some generic task/operation canceled exception. The way to make that determination is, when you catch the operation canceled exception in steps runner, inspect the timeout cancellation token. If so, bubble a TimeoutException from StepsRunner with an appropriate message. Otherwise rethrow.